pkg/services/oauthtoken/oauth_token_test.go

```diff
@@ -264,3 +286,202 @@ func (f *FakeAuthInfoStore) UpdateAuthInfo(ctx context.Context, cmd *login.Updat
+		},
+		{
+			desc:        "should fail if the user identity cannot be converted to an int",
+			identity:    &authn.Identity{ID: "user:invalidIdentifierFormat"},
+			expectedErr: nil,
+		},
+		{
+			desc:     "should skip if the expiration check has been cached",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				cache.Set("oauth-refresh-token-1234", true, 1*time.Minute)
+			},
+			expectedErr: nil,
+		},
+		{
+			desc:     "should skip if the expiration check has been cached",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedError = errors.New("some error")
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+		},
+		{
+			desc:     "should not find the user as the user was not logged via OAuth",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedError = user.ErrUserNotFound
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+		},
+		{
+			desc:     "should refresh the token if the auth module is oauth",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedUserAuth = &login.UserAuth{
+					AuthModule: "oauth",
+				}
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+		},
+		{
+			desc:     "should skip sync with a valid JWT token and uses an unknown idp",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedUserAuth = &login.UserAuth{
+					AuthModule:   "oauth_generic_idp",
+					OAuthIdToken: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
+				}
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+			oauthInfo:            nil,
+		},
+		{
+			desc:     "should skip refresh token if oauth provider token handling is disabled",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedUserAuth = &login.UserAuth{
+					AuthModule:   "oauth_generic_idp",
+					OAuthIdToken: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
+				}
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+			oauthInfo: &social.OAuthInfo{
+				UseRefreshToken: false,
+			},
+		},
+		{
+			desc:     "should skip refresh token if oauth provider token handling is disabled",
```

Comment:
This test currently skips refreshing the token because there is no refresh token

Reply:
This is still true! You need to either fix the test or the description.